File: API/HockeyStat.Model/Model/Game.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HockeyStat.Model.Model
{
    public class Game : IDBEntity
    {
        [Key]
        public long ID { get; set; }

        [Column(TypeName="datetime2")]
        public DateTime Date { get; set; }

        public int HomeScore { get; set; }

        public int GuestScore { get; set; }

        public int OTHomeScore { get; set; }

        public int OTGuestScore { get; set; }

        public int PSHomeScore { get; set; }

        public int PSGuestScore { get; set; }

        [ForeignKey("HomeTeam_ID")]
        public Team HomeTeam { get; set; }

        [ForeignKey("GuestTeam_ID")]
        public Team GuestTeam { get; set; }

        [ForeignKey("Season_ID")]
        public Season Season { get; set; }
    }
}

File: Client/src/model/game.ts
import { Team } from './Team';
import { Season } from './Season';

export class Game {
    id: number;
    date: Date;
    homeScore: number;
    guestScore: number;
    otHomeScore: number;
    otGuestScore: number;
    psHomeScore: number;
    psGuestScore: number;
    homeTeam: Team;
    guestTeam: Team;
    season: Season;

    constructor(date: Date) {
        this.date = date;
        this.guestScore = 0;
        this.homeScore = 0;
        this.otGuestScore = 0;
        this.otHomeScore = 0;
        this.psGuestScore = 0;
        this.psHomeScore = 0;
    }
}

